### PR TITLE
feat: Use system default color mode, but allow user override

### PR DESCRIPTION
**Pull request by ericvolp12**
I've implemented a fix for https://github.com/bluesky-social/social-app/issues/682 using radio buttons and moved the setting to the Settings page as suggested in the comments below:

chrome_GrxyX238hI
chrome_MeFfIhmSL2

The default is currently `system` so new users will be using their device default themes when launching the app for the first time but after changing it, will have their preferences stored in the shell state.

OLD:
I've implemented a fix for https://github.com/bluesky-social/social-app/issues/682 that doesn't use a radio group but cycles through 3 modes `['light', 'dark', 'system']` and the `system` mode inherits the device theme properly.

Video Demo:
chrome_2sS6iwAMYw

When system default theme is dark:
image

When system default theme is light:
image

Regardless of the system default theme:
image
image

Tested in mobile browser as well and it inherits an android device theme properly.

Couldn't get the expo test app to work on my phone yet but I did update the Native components so they should expose the same behavior.

This change doesn't change the `colorScheme` at all, it just updates how the `colorScheme` is set in `ThemeContext.tsx` so that we inherit if the user has chosen `system` as their theme.

We also default in `shell.ts` to use the system theme to conform to user expectations.

### CODE REVIEW

**Comment by ansh** [comment]
I have identical code to this for this issue. The problem is a UX one. It's hard to express to users that a singular icon button is a 3-way toggle between light/dark/system on mobile. It's more verbose on web so easier to understand for sure.

**Comment by ericvolp12** [comment]
I took another whack at it using a 3-state radio button, I can't test on mobile at the moment though.

chrome_7JgJpM8pKt

**Comment by ericvolp12** [comment]
Current mobile web looks like:

chrome_veKI1pZg6A

**Comment by notdaniel** [comment]
> It's hard to express to users that a singular icon button is a 3-way toggle between light/dark/system on mobile.

I think users are used to the word "automatic" in reference to system theme. Paired with the right icon, it would work fine.

So I notice that a moon icon is being used above, even for daytime. If you switched it to a sun icon, then you could use them this way:

- ☀ Light Theme
- ☾ Dark Theme
- ☀/☾ Automatic

I find the text-only buttons to be way less intuitive, personally

**Comment by pfrazee** [comment]
I appreciate what you're aiming for here!

Yeah I think the 3-way state is a bit of a challenge. Maybe we move it out of the drawer/right-nav and into the settings?

**Comment by parion** [comment]
> Maybe we move it out of the drawer/right-nav and into the settings?

That's what I was thinking too, something ala this in a Display menu. 

image

**Comment by snhasani** [comment]
@ericvolp12 I cocked these codes last night. But in the end I wasn't happy about the result because of the poor UI and UX. So I didn't create a PR for here but you can check my code and maybe reusing some part of it. I refactor the SelectableButton component and create a new one, GroupButton. And some other stuff that's [here](https://github.com/snhasani/social-app/pull/1/files).

I think we can put appearance settings into a modal like a few of current settings. Like twitter iOS dark mode settings:

telegram-cloud-photo-size-4-5994759638173727624-y

P.S: Running iOS or Android is a pain in ass. I couldn't run them after a couple of hours of miserable tries. It is too hard to contribute on something that you can not run it :D

cc @pfrazee @ansh

**Comment by ericvolp12** [comment]
Okay, I've moved the appearance 3-way-toggle to the Settings page and removed it from the other components.

chrome_GrxyX238hI
chrome_MeFfIhmSL2

File: src/view/screens/Settings.tsx [hunk 1]
```diff
@@ -282,6 +283,34 @@ export const SettingsScreen = withAuthRequired(
             </Text>
           </TouchableOpacity>
 
+          <View style={styles.spacer20} />
+          <Text type="xl-bold" style={[pal.text, styles.heading]}>
+            Appearance
+          </Text>
+          <View>
+            <View style={[styles.linkCard, pal.view, styles.selectableBtns]}>
```
Comment (snhasani) [inline src/view/screens/Settings.tsx:291]: Here is an opportunity for refactoring and preventing duplicate codes. You can pull out SelectableBtn from `src/view/com/modals/ContentFilteringSettings.tsx` and create a new one component that reused here and there.

You can use my code from [this](https://github.com/snhasani/social-app/commit/7fdf63869b19205b3fa41590a6525491d19709b0) and [this](https://github.com/snhasani/social-app/commit/a818d13f6231485979696bc8f90900490abf7fcb) commit. It's already there.

**Comment by rezaaa** [comment]
Good job guys. I hope you change the component design to something better.

File: src/state/models/ui/shell.ts [hunk 2]
```diff
@@ -189,7 +189,7 @@ export interface ComposerOpts {
 }
 
 export class ShellUiModel {
-  darkMode = false
+  colorMode = 'system'
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply src/state/models/ui/shell.ts:192]: I get this concern but I don't think it's confusing enough to change

**Comment by pfrazee** [comment]
Awesome, thanks everybody

**Pull request merged**